File: python/quicksort.py
def quicksort(arr):
    # that's my base case, array with 1 or 0 are already "sorted"
    if len(arr) < 2:
        return arr
    else:
        # recursive case
        pivot = arr[0]

        # sub-array of all the elements less than the pivot
        less = [i for i in arr[1:] if i <= pivot]

        # sub-array pf all the elements greater than the pivot
        greater = [i for i in arr[1:] if i > pivot]
        return quicksort(less) + [pivot] + quicksort(greater)


arr = [2, 3, 5, 10]
print(quicksort(arr))

File: javascript/quickSort.js
/**
 * Quick array sorting
 * @param {Array} array Source array
 * @returns {Array} Sorted array
 */
console.time(`quicksort`)
function quicksort(array) {
  //base case, array with 0 or 1 element are already sorted
  if (array.length < 2) return array;
  // recursive case
  let pivot = array[0];
  // partitioning the lefts arrays.
  // sub-arrays of all the elements less than the pivot
  let less = array.slice(1).filter(function (element) {
    return element <= pivot;
  })
  //sub-array of all the elements greater than the pivot
  let greater = array.slice(1).filter(function (element) {
    return element > pivot;
  })

  return quicksort(less).concat([pivot], quicksort(greater));
}

let numbersArray = [2, 5, 7, 4, 9, 1];
console.log(quicksort(numbersArray));